mava/types.py

```diff
@@ -21,17 +21,60 @@
 from optax._src.base import OptState
 from typing_extensions import NamedTuple, TypeAlias
 
-from mava.wrappers.jumanji import LogEnvState
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
+
 
 Action: TypeAlias = chex.Array
 Value: TypeAlias = chex.Array
 Done: TypeAlias = chex.Array
 HiddenState: TypeAlias = chex.Array
-
-# Can't know the exact type of State or Timestep.
+# Can't know the exact type of State.
 State: TypeAlias = Any
-Observation: TypeAlias = Any
-RnnObservation: TypeAlias = Tuple[Observation, Done]
+
+
+class Observation(NamedTuple):
+    """The observation that the agent sees.
+    agents_view: the agents' view of other agents and items within their
```

Comment:
Maybe a simpler docstring here is to just say `the agent's view of the environment.`

Reply:
Thanks for suggesting this I'll change it 🙌